libfreerdp-utils/file.c

```diff
@@ -83,6 +83,17 @@ char* freerdp_get_home_path(rdpSettings* settings)
 	if (settings->home_path == NULL)
 		settings->home_path = getenv(HOME_ENV_VARIABLE);
 
+	/* https://github.com/neutrinolabs/xrdp/issues/719 */
+	/* workaround if home_path still is null */
+	/* happens for exmaple if in /lib/systemd/system/xrdp.service the User line is missing: */
+	/* [Service] */
+	/* ... */
+	/* User=root */
+	if(settings->home_path == NULL)
+        {
+                settings->home_path = "/";
```

Comment:
`xstrdup("/");` would be better rather than a raw literal.

See: https://github.com/FreeRDP/FreeRDP/commit/036c8ce47d4400f78c693db63852a5bc95fa2cad

If we change it to `xstrdup("/");`, we're backporting the code from FreeRDP. Maybe copyright notices needed. The main author of FreeRDP Marc-Andre Moreau already appears at the top but this homedir fix is provided by Pawel Jakub Dawidek. We need to record his name at least commit message. Of course in the file will be better.

For example, prefered commit message will be like this.

```
Fix crash neutrinordp when missing HOME env

Imported from: https://github.com/FreeRDP/FreeRDP/commit/036c8ce47d4400f78c693db63852a5bc95fa2cad
Originally fixed by:  Pawel Jakub Dawidek
```